src/components/20-molecules/datepicker/utils/date.js

```diff
@@ -77,9 +84,16 @@ const addLeadingZeroes = (rawNumber, numDigits) => {
   return `${leadingZeroesString}${number}`;
 };
 
-const parseLocalisedDateIfValid = (locale = 'en', inputValue = '') => {
-  if (!Intl.DateTimeFormat.supportedLocalesOf(locale).length) {
-    throw new Error(`locale not supported: ${locale}`);
+const parseLocalisedDateIfValid = (
+  inputLocale = LOCALE_DEFAULT,
+  inputValue = ''
+) => {
+  let locale = inputLocale;
+
+  if (!locale || !Intl.DateTimeFormat.supportedLocalesOf(locale).length) {
```

Comment:
This looks complicated and cumbersome.

Suggestion: extract tests to named variables, like
`const unsupported = !Intl....` and `const missing = !locale;` and `const isSwissItalian = locale === 'it-CH';`

Then do a clean `locale = (missing || unsupported) ? LOCALE_DEFAULT : (isSwissItalian ? LOCALE_DE_CH : locale);`

Reply:
This PR was not ready for review;) But thanks for reviewing! Named variables sounds quite good. Will change that. But i will not use this huge one-liner. This is more complicated than my if.